### PR TITLE
[Kokoro / CI] Have Kokoro test multiple iOS versions and simulators on different Xcodes.

### PR DESCRIPTION
**Pull request by yarneo**
Now Kokoro supports running on a matrix of different iOS versions, simulators, and Xcodes.

We are now imitating the matrix we currently have set up on Travis so we can remove the dependency on it and have a faster testing/CI turnaround for our code pushes and releases.

### CODE REVIEW

File: material_components_ios.bzl [hunk 1]
```diff
@@ -46,3 +47,37 @@ def mdc_public_objc_library(
...
+      name = "xcode8_3_3",
+      values = {"xcode_version": "8.3.3"},
+  )
+
+  native.config_setting(
+      name = "xcode_9_0",
+      values = {"xcode_version": "9.0"},
+  )
+
+  return [":IPHONE_5_IN_8_1", ":IPAD_PRO_12_9_IN_9_3", ":IPHONE_7_PLUS_IN_10_3", ":DYNAMIC_RUNNER"]
```
Comment (jverkoey) [inline material_components_ios.bzl]: Do all of the runners run on each test invocation? Or does blaze identify which config to use based on the command line invocation?

Comment (yarneo) [reply material_components_ios.bzl]: All the runners run on all the components, with the caveat of the DYNAMIC_RUNNER which identifies the argument passed in the blaze invocation (e.g. `bazel -xcode_version 9.1`), and then chooses based on the xcode_version we passed if to create an iPhoneX + iOS11.0 runner or an iPad2 + iOS8.4 for xcode version 8.3.3.

Comment (jverkoey) [reply material_components_ios.bzl]: Cool cool.

**Comment by jverkoey** [comment]
Love the branch name btw :P

**Review by jverkoey — COMMENTED** [review]
Did you already consider making an mdc_unit_test_suite method that implements the default arguments? This might be easier to maintain over time if we need to modify other arguments to the test targets.

**Comment by yarneo** [comment]
That's a great point, I will go ahead and update the PR accordingly.

**Review by ianegordon — APPROVED** [review]
LGTM!
Please merge once the i's are dotted and the t's are crossed.

**Pull request merged**